cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java

```diff
@@ -0,0 +1,174 @@
+                              Bytes.toBytes(GSON.toJson(config.getProperties())));
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void delete(final String namespace, final String type, final String id) throws Exception {
+    txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, Void>() {
+      @Override
+      public Void apply(ConfigTable configTable) throws Exception {
+        if (configTable.table.get(rowKey(namespace, type), Bytes.toBytes(id)) == null) {
+          throw new ConfigNotFoundException(namespace, type, id);
+        }
+        configTable.table.delete(rowKey(namespace, type), Bytes.toBytes(id));
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public List<Config> list(final String namespace, final String type) throws Exception {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, List<Config>>() {
+      @Override
+      public List<Config> apply(ConfigTable configTable) throws Exception {
+        List<Config> configList = Lists.newArrayList();
+        Row row = configTable.table.get(rowKey(namespace, type));
+        for (Map.Entry<byte[], byte[]> col : row.getColumns().entrySet()) {
+          Map<String, String> properties = GSON.fromJson(Bytes.toString(col.getValue()), MAP_STRING_STRING_TYPE);
+          configList.add(new Config(Bytes.toString(col.getKey()), properties));
+        }
+        return configList;
+      }
+    });
+  }
+
+  @Override
+  public Config get(final String namespace, final String type, final String id) throws Exception {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, Config>() {
+      @Override
+      public Config apply(ConfigTable configTable) throws Exception {
+        if (configTable.table.get(rowKey(namespace, type), Bytes.toBytes(id)) == null) {
+          throw new ConfigNotFoundException(namespace, type, id);
+        }
+        byte[] prop = configTable.table.get(rowKey(namespace, type), Bytes.toBytes(id));
+        Map<String, String> propertyMap = GSON.fromJson(Bytes.toString(prop), MAP_STRING_STRING_TYPE);
+        return new Config(id, propertyMap);
+      }
+    });
+  }
+
+  @Override
+  public void update(final String namespace, final String type, final Config config) throws Exception {
+    txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, Void>() {
+      @Override
+      public Void apply(ConfigTable configTable) throws Exception {
+        if (configTable.table.get(rowKey(namespace, type), Bytes.toBytes(config.getId())) == null) {
+          throw new ConfigNotFoundException(namespace, type, config.getId());
+        }
+        configTable.table.put(rowKey(namespace, type), Bytes.toBytes(config.getId()),
+                              Bytes.toBytes(GSON.toJson(config.getProperties())));
+        return null;
+      }
+    });
+  }
+
+  private String rowKeyString(String namespace, String type) {
```

Comment:
heh. So we want to put restrictions on what is allowed in namespace and type ids... Like "no dot"...

I'd rather encode row key as usual with <1-byte-version><4-byte-keyPart1Length><keyPart2Value>[<4-byte-keyPartNLength><keyPartNValue>], where namespace and type are parts